### DISCUSSION TITLE
Use PRESSURE_RELAX_TIME for numerical stability?

### DISCUSSION THREAD

**Discussion by IFAB-AS** [firemodels/fds#13468] [Q&A]
Hello,
I have some model with PRESSURE ZONES (imported vehicle model with BURN_AWAY) and had the idea to use the PRESSURE_RELAX_TIME feature to get around numerical instabilities (MINIMUM_DENSITY / MAXIMUM_DENSITY errors etc.).
PRESSURE_RELAX_TIME gives the time until the pressure zones have the same pressure. I made a simple box with a burning surface inside that is opened when some pressure inside is reached.

What I was hoping for:
Increased PRESSURE_RELAX_TIME reduces the speed of the outflowing gas, which increases numerical stability. It would also emulate deeper layers, that release combustibles more gradually.

What I get:
The velocity at the opening seems to be independent from PRESSURE_RELAX_TIME. But what is FDS changing to reduce the mass flow from the Pressure zone? Is it reducing the density of the outflowing gases? This would explain errors with minimum density.

[hohlraum_relax.txt](https://github.com/user-attachments/files/17110333/hohlraum_relax.txt)

**Comment by drjfloyd** [comment]
Pressure relax time changes how quickly the zone-to-zone pressure difference goes to zero. The flow is based on the pressure difference at each timestep.  Your pressure rise is almost 5 atm, and the critical pressure ratio for air is ~0.5.  At the timestep the zones merge, you are well into choked flow and violating the low Mach number assumption for FDS.

**Comment by mcgratta** [comment]
Make plots of pressure vs time. FDS was not designed for high pressure applications where bursts might occur.

**Comment by IFAB-AS** [comment]
Ok, this means pressure relax time does not affect the initial velocity at the opening when the zone bursts and does not increase stability, only the reduction of the pressure difference or filling up the zones will decrease the velocity.

**Comment by mcgratta** [comment]
I am pretty sure it does. In other words, increasing the relaxation time lowers the flow rate and increases the time to reach an equilibrium pressure. Make some plots of pressure vs time to check this.
